test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java

```diff
@@ -227,30 +228,49 @@ public void testCleanup() throws Exception {
                 .state(),
             equalTo(SnapshotState.SUCCESS));
 
-        logger.info("--> creating a dangling index folder");
         final BlobStoreRepository repo =
             (BlobStoreRepository) getInstanceFromNode(RepositoriesService.class).repository("test-repo");
-        final PlainActionFuture<Void> future = PlainActionFuture.newFuture();
         final Executor genericExec = repo.threadPool().executor(ThreadPool.Names.GENERIC);
+
+        logger.info("--> creating a dangling index folder");
+
+        createDanglingIndex(repo, genericExec);
+
+        logger.info("--> deleting a snapshot to trigger repository cleanup");
+        client().admin().cluster().deleteSnapshot(new DeleteSnapshotRequest("test-repo", snapshotName)).actionGet();
+
+        assertConsistentRepository(repo, genericExec);
+
+        logger.info("--> Create dangling index");
+        createDanglingIndex(repo, genericExec);
+
+        logger.info("--> Execute repository cleanup");
+        final CleanupRepositoryResponse response = client().admin().cluster().prepareCleanupRepository("test-repo").get();
+        assertCleanupResponse(response, 3L, 1L);
+    }
+
+    protected void assertCleanupResponse(CleanupRepositoryResponse response, long bytes, long blobs) {
+        assertThat(response.result().blobs(), equalTo(1L + 2L));
+        assertThat(response.result().bytes(), equalTo(3L + 2 * 3L));
+    }
+
+    private void createDanglingIndex(final BlobStoreRepository repo, final Executor genericExec) throws Exception {
+        final PlainActionFuture<Void> future = PlainActionFuture.newFuture();
         genericExec.execute(new ActionRunnable<>(future) {
             @Override
             protected void doRun() throws Exception {
                 final BlobStore blobStore = repo.blobStore();
-                blobStore.blobContainer(BlobPath.cleanPath().add("indices").add("foo"))
-                    .writeBlob("bar", new ByteArrayInputStream(new byte[0]), 0, false);
+                blobStore.blobContainer(repo.basePath().add("indices").add("foo"))
+                    .writeBlob("bar", new ByteArrayInputStream(new byte[3]), 3, false);
```

Comment:
This whole logic of checking the count and size is kinda meh being hard coded this way I must admit. A better version of this is incoming in the repo cleanup cli PR and I'd just make use of that to clean things up here too once it's merged if that's ok.

Reply:
This PR is already merged, so I think it's time to re-use methods from BlobStoreTestUtils

Reply:
Right ... how time flies :) cleaning this up then :)

Reply:
Hmm turns out those won't work without refactoring after all (they don't properly discriminate between folders and blobs so it's more code to use them than not). I'd just clean this up in a follow up (together with a ton of duplication that we have acquired in these tests anyway) since this PR is already 1k+ lines?